src/webob/acceptparse.py

```diff
@@ -1641,86 +927,54 @@ def __add__(self, other):
           including any media type parameters, and *extension_params* is a
           ``str`` of the extension parameters segment of the header element,
           starting with the first '``;``'
-        * an :class:`AcceptValidHeader`, :class:`AcceptNoHeader`, or
-          :class:`AcceptInvalidHeader` instance
+        * an :class:`Accept` instance
         * object of any other type that returns a value for ``__str__``
 
-        If `other` is a valid header value or an :class:`AcceptValidHeader`
-        instance, then a new :class:`AcceptValidHeader` instance with the valid
-        header value is returned.
-
-        If `other` is ``None``, an :class:`AcceptNoHeader` instance, an invalid
-        header value, or an :class:`AcceptInvalidHeader` instance, a new
-        :class:`AcceptNoHeader` instance is returned.
+        The rules for adding values to a header are that the values are
+        appended if valid, or discarded. If everything is discarded then an
+        instance representing a missing header is returned.
         """
 
-        if isinstance(other, AcceptValidHeader):
-            return AcceptValidHeader(header_value=other.header_value)
-
-        if isinstance(other, (AcceptNoHeader, AcceptInvalidHeader)):
-            return AcceptNoHeader()
+        other = create_accept_header(other)
```

Comment:
This is using `create_accept_header` to handle all the possible types for the value of `other`, but `create_accept_header` is documented as taking a parameter of type `str` or `None`?

Reply:
The add and radd support what they did before - I just moved the logic into the `__init__` as you noted. It sounds like you just want the `__init__` to be documented to take those other types which is fine - I can do that.

Reply:
If you are actually changing `.__init__` to make it accept all those other types of values accepted by `.__add__`/`.__radd__`, then yes, it's only the documentation that needs to be updated.